app/models/concerns/smooch_nlu.rb

```diff
@@ -0,0 +1,108 @@
+require 'active_support/concern'
+
+module SmoochNlu
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    # FIXME: Refactor the rest of the Smooch-related code to use this helper
+    def get_smooch_bot_installation_for_nlu(team_slug)
+      TeamBotInstallation.where(team: Team.find_by_slug(team_slug), user: BotUser.smooch_user).last
+    end
+
+    def toggle_nlu(team_slug, enabled)
+      tbi = self.get_smooch_bot_installation_for_nlu(team_slug)
+      if tbi.nil?
+        false
+      else
+        tbi.set_nlu_menus_enabled = enabled
+        tbi.save!
+        true
+      end
+    end
+
+    def nlu_enabled?
+      !!self.config.to_h['nlu_menus_enabled']
+    end
+
+    def enable_nlu(team_slug)
+      self.toggle_nlu(team_slug, true)
+    end
+
+    def disable_nlu(team_slug)
+      self.toggle_nlu(team_slug, false)
+    end
+
+    def nlu_alegre_id(team_slug, menu, menu_option_id, keyword)
+      Digest::MD5.hexdigest(['smooch-nlu-menu', team_slug, menu, menu_option_id, keyword].join(':'))
+    end
+
+    def common_nlu_params_for_alegre(team_slug, menu, menu_option_id, keyword)
+      {
+        doc_id: self.nlu_alegre_id(team_slug, menu, menu_option_id, keyword),
+        context: {
+          context: 'smooch-nlu-menu',
+          team: team_slug,
+          menu: menu,
+          menu_option_id: menu_option_id
+        }
+      }
+    end
+
+    # FIXME: Make it more flexible
+    def nlu_models_to_use
+      [Bot::Alegre::MEAN_TOKENS_MODEL]
```

Comment:
Very soon I'd like to use paraphrase-ml locally, but until we have that available I wonder if 
`[Bot::Alegre::OPENAI_ADA_MODEL]` is a better choice. Another option would be using both: `[Bot::Alegre::MEAN_TOKENS_MODEL, Bot::Alegre::OPENAI_ADA_MODEL]`. I'd be curious for your subjective experience trying out these options, @caiosba . I can also try locally myself.